samza-yarn/src/main/scala/org/apache/samza/job/yarn/SamzaYarnAppMasterLifecycle.scala

```diff
@@ -44,25 +45,26 @@ class SamzaYarnAppMasterLifecycle(containerMem: Int, containerCpu: Int, samzaApp
   var validResourceRequest = true
   var shutdownMessage: String = null
   var webApp: SamzaYarnAppMasterService = null
-  def onInit() {
+  def onInit(): util.Map[ContainerId, YarnContainer] = {
     val host = state.nodeHost
     val response = amClient.registerApplicationMaster(host, state.rpcUrl.getPort, "%s:%d" format (host, state.trackingUrl.getPort))
 
     // validate that the YARN cluster can handle our container resource requirements
     val maxCapability = response.getMaximumResourceCapability
     val maxMem = maxCapability.getMemory
     val maxCpu = maxCapability.getVirtualCores
+    val previousAttemptContainers = new HashMap[ContainerId, YarnContainer]()
     if (isApplicationMasterHighAvailabilityEnabled) {
       val yarnIdToprocIdMap = new HashMap[String, String]()
       samzaAppState.processorToExecutionId.asScala foreach { entry => yarnIdToprocIdMap.put(entry._2, entry._1) }
       response.getContainersFromPreviousAttempts.asScala foreach { (ctr: Container) =>
         val samzaProcId = yarnIdToprocIdMap.get(ctr.getId.toString)
-        if (samzaProcId != null) {
-          info("Received container from previous attempt with samza processor id %s and yarn container id %s" format(samzaProcId, ctr.getId.toString))
-          samzaAppState.runningProcessors.put(samzaProcId,
-            new SamzaResource(ctr.getResource.getVirtualCores, ctr.getResource.getMemory, ctr.getNodeId.getHost, ctr.getId.toString))
-          state.runningProcessors.put(samzaProcId, new YarnContainer(ctr))
-        }
+        info("Received container from previous attempt with samza processor id %s and yarn container id %s" format(samzaProcId, ctr.getId.toString))
+        samzaAppState.runningProcessors.put(samzaProcId,
+          new SamzaResource(ctr.getResource.getVirtualCores, ctr.getResource.getMemory, ctr.getNodeId.getHost, ctr.getId.toString))
```

Comment:
You don't need to populate the runningProcessors in the new flow right. If you were to populate the `pendingProcessors` instead and with the new flow `YarnClusterResourceManager` invokes a callback on `clusterManagerCallback.onStreamProcessorLaunchSuccess`.
By doing so, you don't need modify any of the `ContainerPlacementManager`'s code and job health, `pendingProcessor` decrement and `runningProcessor` updates should be available for free.